### PR TITLE
Upgrade build process from gulp to webpack

### PR DESCRIPTION
**Pull request by ryanio**
## Description

This PR upgrades the build process from gulp/browserify to webpack.

This PR also:

* Removes `npm run build-all` for creating sub-package browser builds.
  * These builds weren't being published, and we aren't aware of many people using them.
  * They didn't have proper test coverage to ensure integrity of the builds (`web3.min.js` is tested in both chrome and firefox with karma)
  * It is now easy for anyone to create a browser build of a sub-package with this PR. Just modify the root `webpack.config.js` and update the `entry` and `library` to point to the sub-package you would like to build, then run `npm run build`.
* Updates some root deps and removes old/unneeded ones
* Updates readmes `var` => `const`/`let`

## Type of change

- [x] build process enhancements (non-breaking change)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested the resulting files from `dist` folder in a browser.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have tested my code on the live network.
- [x] I have checked the Deploy Preview and it looks correct.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,17 +1,12 @@
 {
     "name": "web3.js",
     "private": true,
-    "version": "1.2.9",
```
Comment (cgewecke) [inline package.json:4]: 👍

**Review by cgewecke — APPROVED** [review]
Looks good! Great improvement 🥇 

Wrt `dist/web3.min.js`...do we usually merge changes to that on release?

**Comment by ryanio** [comment]
@cgewecke thanks! hm yes that makes sense to me, so i can remove it from this PR. edit: done

File: package.json (hunk 1 above)
Comment (GregTheGreek) [reply package.json:4]: why are we removing the version?

File: package.json [hunk 2]
```diff
@@ -20,14 +15,11 @@
         }
     ],
     "scripts": {
-        "version": "npm run build-all",
+        "version": "npm run build",
```
Comment (GregTheGreek) [inline package.json:18]: is this really needed?

File: package.json (hunk 1 above)
Comment (ryanio) [reply package.json:4]: There is no need for the package version in root node_modules as lerna doesn’t use it and doesn’t update it with `lerna version`

File: package.json (hunk 2 above)
Comment (ryanio) [reply package.json:18]: Yes this lifecycle hook creates, commits, and tags the new build with the updated package version numbers as part of running `lerna version`

To learn more see  https://github.com/lerna/lerna/tree/master/commands/version#lifecycle-scripts

**Review by GregTheGreek — APPROVED** [review]

**Pull request merged**